indexing-service/src/main/java/org/apache/druid/indexing/worker/WorkerTaskManagerMonitor.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.worker;
+
+import com.google.inject.Inject;
+import org.apache.druid.indexing.common.task.Task;
+import org.apache.druid.java.util.emitter.service.ServiceEmitter;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.apache.druid.java.util.metrics.AbstractMonitor;
+import org.apache.druid.query.DruidMetrics;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+public class WorkerTaskManagerMonitor extends AbstractMonitor
+{
+  private final WorkerTaskManager workerTaskManager;
+  private static final String WORKER_RUNNING_TASK_COUNT_METRICS = "worker/task/running/count";
+  private static final String WORKER_ASSIGNED_TASK_COUNT_METRIC = "worker/task/assigned/count";
+  private static final String WORKER_COMPLETED_TASK_COUNT_METRIC = "worker/task/completed/count";
+
+  @Inject
+  public WorkerTaskManagerMonitor(WorkerTaskManager workerTaskManager)
+  {
+    this.workerTaskManager = workerTaskManager;
+  }
+
+  @Override
+  public boolean doMonitor(ServiceEmitter emitter)
+  {
+    final Map<String, Integer> runningTasks, assignedTasks, completedTasks;
+
+    runningTasks = getDataSourceTasks(workerTaskManager.getRunningTasks(), WorkerTaskManager.TaskDetails::getDataSource);
+    assignedTasks = getDataSourceTasks(workerTaskManager.getAssignedTasks(), Task::getDataSource);
+    completedTasks = getDataSourceTasks(workerTaskManager.getCompletedTasks(), TaskAnnouncement::getTaskDataSource);
```

Comment:
Preferred style in Druid (if not Java itself):
```suggestion
    final Map<String, Integer> runningTasks = getDataSourceTasks(workerTaskManager.getRunningTasks(), WorkerTaskManager.TaskDetails::getDataSource);
    final Map<String, Integer> assignedTasks = getDataSourceTasks(workerTaskManager.getAssignedTasks(), Task::getDataSource);
    final Map<String, Integer> completedTasks = getDataSourceTasks(workerTaskManager.getCompletedTasks(), TaskAnnouncement::getTaskDataSource);
```